tests/framework/scheduler.py

```diff
@@ -6,12 +6,27 @@
 user to choose the maximum number of worker processes that can run tests
 concurrently.
 
-Tests are split into batches, each batch being assigned a maximum concurrency
-level. For instance, all performance tests will run sequentially
-(i.e. concurrency=1), since they rely on the availability of the full host
-resources, in order to make accurate measurements. Additionally, other tests
-may be restricted to running sequentially, if they are per se
-concurrency-unsafe. See `PytestScheduler.pytest_runtestloop()`.
+Tests are typically run one at a time, with some tests being executed in
+parallel with others.
+There is an assumption made, that if a test starts a microvm, its performance
+may be affected by other tests running in parallel with it, so for this reason
+the test framework limits tests that launch microvms to be executed in parallel
+with others.
+
+In order to check if a test launches a VM, the framework looks up what fixtures
+the test uses. If one of those fixtures matches the list enumerated in
+`PytestScheduler.TestItem.CONC_FIXTURES_PREF`, then the tests concurrency limit
+will be set to 1, which means that while that test is running, at most one test
+can run in parallel with it.
+
+Although these limitations are detected automatically, one can also define
+concurrency limits manually by adding the `@pytest.mark.concurrency(ARG)`
+decorator, where ARG can be a number or 'max'.
+If ARG is a number, that number will be set as the concurrency level for the
+test. If ARG is 'max' then the test will run using the maximum concurrency
+limit set.
+If ARG > maximum concurrency limit, then the concurrency level will not exceed
```

Comment:
The concurrency limit should probably be a constant somewhere and this doc block should reference it.

Reply:
Having looked over the code some more, we should just say that this limit is given to the `--concurrency` CLI param